### ISSUE TITLE
[android] My location sharing crash

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#1822]
Latest master 2022.01.09-11-FDroid-debug
Xiaomi Mi A2 Lite, custom Android 9.0
Works fine (no crash) on latest stable 2021.12.01-4-FDroid

Share your own location (via menu button or by selecting the location arrow and pressing Share), choose some sharing option that will cover the screen fully (i.e. not a pop-up), OM process should go into background. Wait half a minute, return to OM by pressing Back. OM app screen shows app, then crashes.

I couldn't reproduce it in the emulator (Android 5 & 11), maybe because there are no sharing options that make it crash.
Before the crash I see in the logs `D/LocationHelper: removeListener(), listener: LocationHelper.mCoreLocationListener`. If this string didn't appear then it wouldn't crash, it looks like.

P.S. #1821 is not involved

[myloc-share-crash.log](https://github.com/organicmaps/organicmaps/files/7835812/myloc-share-crash.log)

**Comment by vng** [comment]
Debug assert fires. Stable FDroid is not fired because it's built like release.
Try comment ASSERT here frontend_renderer.cpp:1456, rebuild and check how it works in debug then.

**Comment by pastk** [comment]
With commented out assert it works fine (doesn't crash).

What's the point of having this assert if it fires in normal situations? Let's remove it?
E.g. when I was testing #1821 I thought the crashes are due to my changes and wasted some time debugging.

**Comment by biodranik** [comment]
@vng Это баг состояния. Drape дальше пытается рисовать эту позицию:
```
        ASSERT(m_myPositionController->IsModeHasPosition(), ());
        m_selectionShape->SetPosition(m_myPositionController->Position());
        m_selectionShape->Render(m_context, make_ref(m_gpuProgramManager), modelView,
                                 m_currentZoomLevel, m_frameValues);
```

Надо разобраться, почему глючит состояние, может быть наши правки локейшна сломали логику. Смущает в логе отбрасывание "менее точной позиции", может этот кейз как раз и плохо обрабатывается у нас.

**Comment by pastk** [comment]
Дополнение:
падает и с gps fix (blue arrow), и без (grey arrow)

**Comment by biodranik** [comment]
Да, креш стабильный в дебаге. Не надо долго ждать, достаточно просто перейти в другую аппу для шаринга и сразу вернуться обратно.

**Comment by biodranik** [comment]
Ноги растут из-за вот этого кода, который сбрасывает текущий режим с позицией в NotFollowNoPosition:
```c++
void MyPositionController::OnEnterBackground()
{
  if (!m_isInRouting && !df::IsModeChangeViewport(m_mode))
    ChangeMode(location::NotFollowNoPosition);
}
```
